File: 프로그래머스/2/42746. 가장 큰 수/가장 큰 수.py
def solution(numbers):
    numbers = list(map(str, numbers))
    numbers.sort(key = lambda x : x*3, reverse = True)
    
    return str(int(''.join(numbers))) # [0,0] 이라면 답이 "00"이 아닌 "0"어야 함

File: 프로그래머스/2/42746. 가장 큰 수/가장 큰 수.java
import java.util.*;

class Solution {
    public String solution(int[] numbers) {
        String answer = "";
        String[] strNums = new String[numbers.length];
        
        for (int i = 0; i < numbers.length; i++)
            strNums[i] = String.valueOf(numbers[i]);
        
        Arrays.sort(strNums, (a, b) -> (b + a).compareTo(a + b));
        
        answer = String.join("", strNums);
        
        if (answer.charAt(0) == '0')
            return "0";
        
        return answer;
    }
}